### PR TITLE
✨ Introduce `paths`-option Syntax to specify Array Content Persistance

### PR DESCRIPTION
**Pull request by floriankapaun**
## Description

Enables developers to use `'[]'` in the [`paths`-option](https://prazdevs.github.io/pinia-plugin-persistedstate/guide/config.html#paths), to specify the persistance of all of an arrays contents.

Example:

```ts
import { defineStore } from 'pinia'

export const useStore = defineStore('store', {
  state: () => ({
    save: [
      { me: 'saved', ignored: 'ignored' }
      { me: 'saved-too', value: 'ignored-too' }
      { me: 'saved-as-well', test: 'ignored-as-well' }
    ],
  }),
  persist: {
    paths: ['save.[].me'],
  },
})
```

In this store, only `save[0].me`, `save[1].me` and `save[2].me` values will be persisted.

## Linked Issues

implements https://github.com/prazdevs/pinia-plugin-persistedstate/issues/116

### CODE REVIEW

File: src/core/pick.ts [hunk 1]
```diff
@@ -1,24 +1,134 @@
 import type { StateTree } from 'pinia'
+import type { Mergeable } from '~/core/types'
+
+function merge(destination: Mergeable, source: Mergeable): Mergeable {
+  const mergingArrays = Array.isArray(destination) && Array.isArray(source)
+  const mergingObjects =
```
Comment (prazdevs) [inline src/core/pick.ts:6]: can you create a small `function isObject(value: unknown) {}` util for readability concern ?

File: src/core/pick.ts [hunk 2]
```diff
@@ -1,24 +1,134 @@
...
+  if (!mergingArrays && !mergingObjects) {
+    throw new Error('Can only merge object with object or array with array')
+  }
+
+  const result = (mergingArrays ? [] : {}) as Mergeable
+
+  const keys: string[] = [...Object.keys(destination), ...Object.keys(source)]
+  keys.forEach((key: string) => {
+    if (Array.isArray(destination[key]) && Array.isArray(source[key])) {
+      return (result[key] = [
```
Comment (prazdevs) [inline src/core/pick.ts:21]: maybe avoid returning when the function is a `=> void` function and use if/else if instead ? 
not a fan of returning assignments

**Review by prazdevs — CHANGES_REQUESTED** [review]
besides changes, it seems interesting and working, encountered any edge case ?

File: src/core/pick.ts (hunk 2 above)
Comment (floriankapaun) [reply src/core/pick.ts:21]: Yes, very good point. Thanks for bringing this to my attention. Will improve this for sure. 🙂💪

**Comment by floriankapaun** [comment]
> besides changes, it seems interesting and working, encountered any edge case ?

Had to rewrite my implementation a few times, because of unexpected hurdles. But no more unhandled edge cases left now – as far as I can tell. 🙂

**Review by prazdevs — APPROVED** [review]

**Comment by prazdevs** [comment]
Seems good, I'll merge it and test  some things then eventually release it as part of 2.2 🎉
